src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState, ...action.extraData,
+        };
+        break;
+      case 'removeProperties':
+        currentState = { ...currentState };
+        action.keysToRemove.forEach(key => delete currentState[key]);
```

Comment:
You don't need to reassign currentState
```suggestion
        action.keysToRemove.forEach(key => delete currentState[key]);
```